public/v4/apps/invoiceninja.yml

```diff
@@ -0,0 +1,150 @@
+          validRegex: /.{1,}/
+
+        - id: $$cap_invoiceninja_DB_DATABASE
+          label: DB_DATABASE
+          defaultValue: ninja
+          validRegex: /^([a-zA-Z0-9\-\.\_])+$/
+
+        - id: $$cap_invoiceninja_DB_ROOT_PASSWORD
+          label: DB_ROOT_PASSWORD
+          defaultValue: $$cap_gen_random_hex(10)
+          description: 'The password of DB superuser.'
+          validRegex: /.{1,}/
+
+        - id: $$cap_invoiceninja_debugmode
+          label: Debug Mode
+          defaultValue: 'false'
+          description: 'Useful for debugging, not suitable for production'
+          validRegex: /^(true|false)$/
+
+services:
+    $$cap_appname-app:
+        restart: always
+        depends_on:
+            - $$cap_appname-db
+        user:
+        environment:
+            APP_URL: http://$$cap_appname.$$cap_root_domain
+            APP_KEY: $$cap_invoiceninja_APP_KEY
+            APP_DEBUG: $$cap_invoiceninja_debugmode
+            REQUIRE_HTTPS: false
+            PHANTOMJS_PDF_GENERATION: false
+            PDF_GENERATOR: snappdf
+            QUEUE_CONNECTION: database
+            DB_HOST: srv-captain--$$cap_appname-db
+            DB_PORT: 3306
+            DB_DATABASE: $$cap_invoiceninja_DB_DATABASE
+            DB_USERNAME: $$cap_invoiceninja_DB_USERNAME
+            DB_PASSWORD: $$cap_invoiceninja_DB_PASSWORD
+            # TODO defined as variables
+            IN_USER_EMAIL: $$cap_invoiceninja_admin_user_email
+            IN_PASSWORD: $$cap_invoiceninja_admin_pass
+            TRUSTED_PROXIES: 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
+        volumes:
+            - $$cap_appname-app-public:/var/www/app/public:rw
+            - $$cap_appname-app-storage:/var/www/app/storage:rw
+        caproverExtra:
+            notExposeAsWebApp: true
+            dockerfileLines:
+                - FROM invoiceninja/invoiceninja:$$cap_invoiceninja_VERSION
+                - USER root
+                - RUN mkdir -p /var/www/app/public /var/www/app/storage
+                - RUN chown 1500:1500 /var/www/app/*
+                - USER 1500
+
+    $$cap_appname-db:
+        restart: always
+        image: mariadb:10.4
+        environment:
+            MYSQL_ROOT_PASSWORD: $$cap_invoiceninja_DB_ROOT_PASSWORD
+            MYSQL_USER: $$cap_invoiceninja_DB_USERNAME
+            MYSQL_PASSWORD: $$cap_invoiceninja_DB_PASSWORD
+            MYSQL_DATABASE: $$cap_invoiceninja_DB_DATABASE
+        volumes:
+            - $$cap_appname-db:/var/lib/mysql:rw
+        caproverExtra:
+            notExposeAsWebApp: true
+
+    $$cap_appname-server:
+        restart: always
+        depends_on:
+            - $$cap_appname-app
+        environment:
+            APP_CONTAINER: srv-captain--$$cap_appname-app:9000
+        volumes:
+            - $$cap_appname-app-public:/var/www/app/public:ro
+        caproverExtra:
+            # Workaround for hard-coded container name in nginx config
+            dockerfileLines:
+                - FROM nginx
```

Comment:
Add a version tag here. No need to export it in variables, but at least we should fix this version tag.

Reply:
fixed to 1.21